core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1059,9 +1048,14 @@ public void parseXContent(QueryParseContext context, AggregatorParsers aggParser
                             docValueFields.add(parser.text());
                         } else {
                             throw new ParsingException(parser.getTokenLocation(), "Expected [" + XContentParser.Token.VALUE_STRING +
-                                    "] in [" + currentFieldName + "] but found [" + token + "]", parser.getTokenLocation());
+                                "] in [" + currentFieldName + "] but found [" + token + "]", parser.getTokenLocation());
                         }
                     }
+                } else if (context.getParseFieldMatcher().match(currentFieldName, INDICES_BOOST_FIELD)) {
+                    indexBoosts = new ArrayList<>();
```

Comment:
rather than re-initializing the list, shall we enforce the fact that users cannot specify boost in both ways? Either deprecated or new syntax, not both at the same time?